MediaAdminBundle/Tests/Form/Type/OrchestraSiteForFolderChoiceTypeTest.php

```diff
@@ -0,0 +1,143 @@
+
+        $this->groupC = Phake::mock('OpenOrchestra\GroupBundle\Document\Group');
+        Phake::when($this->groupC)->getSite()->thenReturn($this->siteDeleted);
+        Phake::when($this->groupC)->hasRole('ROLE_ACCESS_TREE_FOLDER')->thenReturn(true);
+
+        $this->user = Phake::mock('OpenOrchestra\UserBundle\Document\User');
+        Phake::when($this->user)->getGroups()->thenReturn(array($this->groupA, $this->groupB, $this->groupC));
+
+        $this->token = Phake::mock('Symfony\Component\Security\Core\Authentication\Token\TokenInterface');
+        Phake::when($this->token)->getUser()->thenReturn($this->user);
+
+        $this->tokenStorage = Phake::mock('Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface');
+        Phake::when($this->tokenStorage)->getToken()->thenReturn($this->token);
+
+        $this->embedSiteToSiteIdTransformer = Phake::mock('OpenOrchestra\MediaAdminBundle\Form\DataTransformer\EmbedSiteToSiteIdTransformer');
+
+        $this->form = new OrchestraSiteForFolderChoiceType(
+            $this->siteRepository,
+            $this->tokenStorage,
+            $this->embedSiteToSiteIdTransformer
+        );
+
+        $this->choiceList = array($this->siteId1 => $this->siteName1);
+    }
+
+    /**
+     * Test Name
+     */
+    public function testName()
+    {
+        $this->assertSame('orchestra_site_for_folder_choice', $this->form->getName());
+    }
+
+    /**
+     * Test Parent
+     */
+    public function testParent()
+    {
+        $this->assertSame('choice', $this->form->getParent());
+    }
+
+    /**
+     * Test buildForm with 'embed' option
+     */
+    public function testBuildFormWithEmbedOption()
+    {
+        $builder = Phake::mock('Symfony\Component\Form\FormBuilder');
+
+        $this->form->buildForm($builder, array('embed' => true));
+
+        Phake::verify($builder)->addModelTransformer($this->embedSiteToSiteIdTransformer);
+    }
+
+    /**
+     * Test buildForm without 'embed' option
+     */
+    public function testBuildFormWithoutEmbedOption()
+    {
+        $builder = Phake::mock('Symfony\Component\Form\FormBuilder');
+
+        $this->form->buildForm($builder, array('embed' => false));
+
+        Phake::verify($builder, Phake::never())->addModelTransformer($this->embedSiteToSiteIdTransformer);
+    }
+
+    /**
+     * Test the default options
+     */
+    public function testConfigureOptions()
+    {
+        $resolverMock = Phake::mock('Symfony\Component\OptionsResolver\OptionsResolver');
+
+        $this->form->configureOptions($resolverMock);
+
+        Phake::verify($resolverMock)->setDefaults(array(
+                'embed' => false,
+                'choices' => $this->choiceList
+        ));
+    }
+
+    /**
+     * Test Choices list creation 
+     */
+    public function testGetChoices()
+    {
+        $this->assertSame($this->choiceList, $this->form->getChoices());
```

Comment:
It's no more a unit test if you're testing a method not directly but by testing the result of another which may probably (or not) use it. No ?